File: apps/admin/lib/medication-plan.ts
export type MedicationPlanItem = {
  dose: string;
  key: string;
  label: string;
  notes?: string;
  prn?: boolean;
  total: number;
};

export type MedicationPlanGroup = {
  key: string;
  label: string;
  meds: MedicationPlanItem[];
};

export const MEDICATION_PLAN: MedicationPlanGroup[] = [
  {
    key: "morning",
    label: "Morning",
    meds: [
      {
        dose: "morning dose",
        key: "propranolol_morning",
        label: "Propranolol",
        total: 1,
      },
      {
        dose: "30mg XR",
        key: "adderall_xr",
        label: "Adderall XR",
        notes: "Once in the morning.",
        total: 1,
      },
      {
        dose: "2 pills with food",
        key: "multivitamin",
        label: "Multivitamin",
        notes: "Keep roughly one hour away from Adderall.",
        total: 2,
      },
    ],
  },
  {
    key: "afternoon",
    label: "Afternoon",
    meds: [
      {
        dose: "20mg IR",
        key: "adderall_ir",
        label: "Adderall IR",
        notes: "Once in the afternoon.",
        total: 1,
      },
      {
        dose: "3 pills",
        key: "omega_3",
        label: "Omega 3",
        notes: "Fish oil target: 1300mg EPA / 860mg DHA.",
        total: 3,
      },
    ],
  },
  {
    key: "night",
    label: "Night",
    meds: [
      {
        dose: "night dose",
        key: "propranolol_night",
        label: "Propranolol",
        total: 1,
      },
      {
        dose: "2 x 100mg",
        key: "magnesium",
        label: "Magnesium",
        notes: "Before sleeping.",
        total: 2,
      },
      {
        dose: "night / anxiety",
        key: "hydroxyzine",
        label: "Hydroxyzine",
        notes: "As-needed logging only; untouched is not counted as missed.",
        prn: true,
        total: 0,
      },
    ],
  },
];

export const MEDICATION_ITEMS = MEDICATION_PLAN.flatMap((group) => group.meds);

File: packages/core/medication_plan.py
"""Shared medication/supplement schedule metadata for admin and PDF state.

Historical dose observations live as ``SourceEvent`` records. This module only
contains display metadata so rendered month state can show the current regimen.
"""

from __future__ import annotations

from packages.core.models import MedicationGroup, MedicationItem


def default_medication_groups() -> list[MedicationGroup]:
    return [
        MedicationGroup(
            key="morning",
            label="Morning",
            meds=[
                MedicationItem(
                    dose="morning dose",
                    key="propranolol_morning",
                    label="Propranolol",
                    short="Pro",
                    total=1,
                ),
                MedicationItem(
                    dose="30mg XR",
                    key="adderall_xr",
                    label="Adderall XR",
                    short="XR",
                    total=1,
                ),
                MedicationItem(
                    dose="2 pills with food",
                    key="multivitamin",
                    label="Multivitamin",
                    short="MV",
                    total=2,
                ),
            ],
        ),
        MedicationGroup(
            key="afternoon",
            label="Afternoon",
            meds=[
                MedicationItem(
                    dose="20mg IR",
                    key="adderall_ir",
                    label="Adderall IR",
                    short="IR",
                    total=1,
                ),
                MedicationItem(
                    dose="3 pills",
                    key="omega_3",
                    label="Omega 3",
                    short="Ω3",
                    total=3,
                ),
            ],
        ),
        MedicationGroup(
            key="night",
            label="Night",
            meds=[
                MedicationItem(
                    dose="night dose",
                    key="propranolol_night",
                    label="Propranolol",
                    short="Pro",
                    total=1,
                ),
                MedicationItem(
                    dose="2 x 100mg",
                    key="magnesium",
                    label="Magnesium",
                    short="Mg",
                    total=2,
                ),
                MedicationItem(
                    dose="night / anxiety",
                    key="hydroxyzine",
                    label="Hydroxyzine",
                    short="Hyd",
                    prn=True,
                    total=0,
                ),
            ],
        ),
    ]
